### PR TITLE
fully separate view implementation from PentagoLib

### PR DESCRIPTION
**Pull request by Youw**

### CODE REVIEW

**Comment by Youw** [comment]
This also fixes a Linux build

**Review by GasperPaul — CHANGES_REQUESTED** [review]
I'm ok with this conceptually.

My only concern right now is why are we having this completely weird folder structure? There are two different lib folders (`PentagoLib` and `pentagolib`). I still don't follow the logic for that separation. Can we have that discussed and sorted in this PR?

**Comment by Youw** [comment]
This PR is also about renaming `pentagolib`->`PentagoLib`.

So we have:
```
PentagoLib /
    include/PentagoLib
        ...// some .h file
    ... // some .cpp files
    ... // some private headers <not in our case - not yet>
```

It is a known practice, to separate your public headers to a separate folder (in our case `include/PentagoLib`). So it gives us:
- ability to specify `include` folder as include path for your target, and separate your private headers, if there are any; (there are also other advantages, that I can come up with)
- now the users of this library/target has to explicitly specify library, where specific header should be found: `#include <PentagoLib/Game.h>`; (remember how boost does this: `#include <boost/something.hpp>`)

I think this one of the possibles approaches for project structure, as an opposite to simple "all cpp/h in a single folder" (which has other advantages sometime).

**Comment by GasperPaul** [comment]
> This PR is also about renaming `pentagolib`->`PentagoLib`.

Now, I'm pro all of that. I, personally, would prefer lowercase, but it's beside the point. I am more worried about why exactly we have the following folder structure at a higher level:

    src /
        PentagoLib /
            include /
                <all the header>
            src /
                Game.cpp
            test
        WidgetsApp
        pentagolib / 
            src /
                <all other cpp>
            test
            CMakeLists.txt

I have a few issues with this, main being why are we have thouse two top level different case versions of pentagolib folder, one with headers and one cpp, other with all the cpps. It's clearly either a bad name choosen for one (because OS differences?), or a case of not moving all the files. They existed before this PR, but I think we should fix it now, while we're at it.

I propose we move everything into PentagoLib and be done with this. Or am I missing something?

**Comment by GasperPaul** [comment]
I would also argue that we should have either

    PentagoLib / include
        / src
    WidgetsApp / include
        / src

or
    src / PentagoLib
        / WidgetsApp
    include / PentagoLib
        / WidgetsApp

But this is not the main contention point for me.

**Comment by Youw** [comment]
Your second suggestion has a huge design flaw - you're mixing sources of two separate libraries, which is conceptually a bad idea.

In the first option we're loosing the ability to separate "headers namespace" for different libraries.
E.g., right now we can:

    #include <PentagoLib/Game.h>

With you suggestion - it is not possible.

There is no reason to keep *.h and *.cpp files in a separate folder, for "just to keep them separate",
so practically, `PentagoLib/include` + `PentagoLib/src` is no better than `PentagoLib/<everything is here>`.
But putting all public headers under `PentagoLib` sub-folder gives us a powerful advantage: now we can have `#include <PentagoLib/Board.h>`  and (yes, I'm making this up, just get the idea)`#include <ChessLib/Board.h>` in the same .cpp file without any name conflicts or unexpected compiler errors.

@sersajur, what's you opinion?

**Comment by Youw** [comment]
Removing tom-most `src` is not a good idea.
It separates our own "the code" from everything else (3d-party libraries, top-most build scripts, tools, deployer scripts, etc.).

> In scenario one, we add public headers in `PentagoLib/include`

And you no longer can have something like `#include <PentagoLib/Game.h>`.
Either `#include <Game.h>`, either `#include <PentagoLib/include/Game.h>` which looks so weird.

Also, don't forget, that you shouldn't have access to headers of the target, that is not explicitly listed as "required" by your current project.

**Comment by Youw** [comment]
Finally get rid of lower case library. I didn't see it exists.

**Review by GasperPaul — APPROVED** [review]

File: src/PentagoLib/test/main.cpp [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#include <gtest/gtest.h>
+
+
+int main(int argc, char* argv[])
```
Comment (sersajur) [inline src/PentagoLib/test/main.cpp]: This file is unused, we use predefined by googletest main file.

**Review by sersajur — COMMENTED** [review]
Good changes, I like the way you choose naming.

**Review by sersajur — APPROVED** [review]

**Pull request merged**